spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskDeleteService.java

```diff
@@ -216,7 +216,16 @@ public void deleteTaskExecutions(Set<Long> taskExecutionIds) {
 		auditData.put("Deleted Job Execution IDs", StringUtils.collectionToDelimitedString(jobExecutionIds, ", "));
 
 		if (!jobExecutionIds.isEmpty()) {
-			final int numberOfDeletedBatchStepExecutionContextRows = dataflowJobExecutionDao.deleteBatchStepExecutionContextByJobExecutionIds(jobExecutionIds);
+			final Set<Long> stepExecutionIds = dataflowJobExecutionDao.findStepExecutionIds(jobExecutionIds);
+
+			final int numberOfDeletedBatchStepExecutionContextRows;
+			if (!stepExecutionIds.isEmpty()) {
```

Comment:
Why not use `stepExecutionIds.size()`?

Reply:
With `size()` I would need to check `stepExecutionIds.size() > 0`. I usually prefer the `!stepExecutionIds.isEmpty()` notation. If you feel strongly I can change.